Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -3328,16 +3331,22 @@ bool RenderLayerCompositor::requiresCompositingForPlugin(RenderLayerModelObject&
     IntRect contentBox = snappedIntRect(pluginRenderer.contentBoxRect());
     return (contentBox.height() * contentBox.width() > 1);
 }
-    
-bool RenderLayerCompositor::requiresCompositingForFrame(RenderLayerModelObject& renderer, RequiresCompositingData& queryData) const
+
+bool RenderLayerCompositor::isHiddenSubframe(RenderLayerModelObject& renderer) const
 {
     if (!is<RenderWidget>(renderer))
         return false;
 
     auto& frameRenderer = downcast<RenderWidget>(renderer);
-    if (frameRenderer.style().visibility() != Visibility::Visible)
+    return frameRenderer.style().visibility() != Visibility::Visible;
```

Comment:
Does an iframe inside a `visibility:hidden` ancestor get composited? Is there code somewhere else that prevents that?

We should also test a `visibility:visible` iframe in a `visibility:hidden` ancestor.

Reply:
I added the test (composited-iframe-visibility-hidden-ancestor.html).